python/mxnet/gluon/contrib/estimator/estimator.py

```diff
@@ -269,9 +271,14 @@ def evaluate(self,
 
         for metric in val_metrics:
             metric.reset()
+        estimator_ref = self
 
         for _, batch in enumerate(val_data):
-            self.evaluate_batch(batch, val_metrics, batch_axis)
+            if self.estimator_model is None:
+                self.evaluate_batch(batch, val_metrics, batch_axis)
```

Comment:
This line provides backwards compatibility to the unreleased API introduced in https://github.com/apache/incubator-mxnet/pull/16678/ . As that API has not been released yet, we don't need to provide compatibility for it.
Instead it may better to keep the code simple by removing `evaluate_batch` and `fit_batch` from the `Estimator` and only using the plug-and-play version. What do you think?

Reply:
I think wrapping the `fit_batch` and `evaluate_batch` in one class is better. But I will brainstorm a better class name for the two methods.

Reply:
Agree, but the point here is about the `Estimator.fit_batch` and `Estimator.evaluate_batch`. Currently there's a code duplication with `EstimatorModel.fit_batch` etc

Reply:
I have removed the duplicate code for `fit_batch` and `evaluate_batch`.